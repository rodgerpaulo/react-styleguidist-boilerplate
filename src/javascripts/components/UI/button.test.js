import React from 'react';
import Button from './button';
import renderer from 'react-test-renderer';

test('Render Button with rounded border', () => {
  const component = renderer.create(<Button>click me!</Button>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Button without rounded border', () => {
  const component = renderer.create(
    <Button roundedBorder={false}>click me!</Button>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Button with type submit', () => {
  const component = renderer.create(<Button type="submit">Submit form</Button>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
