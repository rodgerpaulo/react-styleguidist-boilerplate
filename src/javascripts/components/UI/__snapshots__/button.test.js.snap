// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button with rounded border 1`] = `
<button
  className="btn sans-bold default rounded"
  onClick={undefined}
  type="button"
>
  click me!
</button>
`;

exports[`Render Button with type submit 1`] = `
<button
  className="btn sans-bold default rounded"
  onClick={undefined}
  type="submit"
>
  Submit form
</button>
`;

exports[`Render Button without rounded border 1`] = `
<button
  className="btn sans-bold default"
  onClick={undefined}
  type="button"
>
  click me!
</button>
`;
